src/assets/thirdPartyServices/snet/news_summary/index.js

```diff
@@ -0,0 +1,96 @@
+import React from "react";
+import  {TextSummary} from  "./summary_pb_service";
+
+export default class NewSummaryService extends React.Component {
+  constructor(props) {
+    super(props);
+    this.submitAction = this.submitAction.bind(this);
+    this.handleFormUpdate = this.handleFormUpdate.bind(this);
+
+    this.state = {
+      serviceName: "TextSummary",
+      methodName: "summary",
+      article_content:
+        'Analysts are predicting record highs as a global shortage of teddy bears sweeps the nation. "The market these products is way up". The advice is to stay indoors as society collapses under the demand.',
+       isComplete: false,
+    };
+  }
+
+  canBeInvoked() {
+    return this.state.text !== "";
+  }
+
+  handleFormUpdate(event) {
+    this.setState({
+      [event.target.name]: event.target.value,
+    });
+  }
+
+  submitAction() {
+    const { methodName, article_content } = this.state;
+    const methodDescriptor = TextSummary[methodName];
+    const request = new methodDescriptor.requestType();
+
+    request.setArticleContent(article_content)
+    //request.setModel(model);
+
+
+    const props = {
+        request,
+        onEnd: ({ message }) => {
+          this.setState({ isComplete: true, response: { value: message.getArticleSummary() } });
```

Comment:
The state `article_content` and `methodName` could be extracted out as `initalUserInput` and reset here..It would be vital for a feature `reset and run` in the new DApp..
you could refer to this commit for the changes.
https://github.com/Vivek205/refactored-dapp/commit/ceb8036cb3ca1b1e6c7153a110d679b21f1f874d